pallets/omnipool/src/traits.rs

```diff
@@ -143,17 +143,28 @@ where
 			return Ok(());
 		}
 
-		let external_price = ExternalOracle::get_price(asset_a, asset_b).map_err(|_| ())?;
-		let external_price = FixedU128::checked_from_rational(external_price.n, external_price.d).ok_or(())?;
 		let max_allowed = FixedU128::from(MaxAllowed::get());
-		let max_allowed_difference = external_price.saturating_mul(max_allowed);
+
+		let oracle_price = ExternalOracle::get_price(asset_a, asset_b).map_err(|_| ())?;
+		let external_price = FixedU128::checked_from_rational(oracle_price.n, oracle_price.d).ok_or(())?;
 		let current_spot_price = FixedU128::checked_from_rational(current_price.n, current_price.d).ok_or(())?;
 
+		let max_allowed_difference = max_allowed
+			.checked_mul(
+				&current_spot_price
+					.checked_add(&external_price)
```

Comment:
I don't get it, why average the two?

Reply:
we need to calculate percentage difference between 2 values. using formula 

`| oracle price - spot price } / ( (oracle price + spot price )  / 2 ) `

Reply:
> Ok, I feel like I'm getting the gist. Can you just add a test that compares this function to a simple annotated calculation?

added simple testcases